libraries/AP_TTLServo/AP_TTLServo.cpp

```diff
@@ -0,0 +1,388 @@
+                     be within the position control range of the servos, normally
+                     0 to 4095
+       @Range: 0 4095
+       @User: Standard */
+    AP_GROUPINFO("POSMAX", 3, AP_TTLServo, pos_max, 4095),
+
+    /* @Param: DESSPD
+       @DisplayName: Servo desired running speed
+       @Description: Value of the desired running speed of the servo. Value and
+                     units are servo dependent, see servo datasheet
+       @Range: 0 65535
+       @User: Standard */
+    AP_GROUPINFO("DESSPD", 4, AP_TTLServo, servo_des_run_speed, RUNNING_SPEED),
+
+    /* @Param: SPDREG
+       @DisplayName: Servo running speed register address
+       @Description: Address of the servo's running speed register, to which the
+                     desired running speed will be written to
+       @Range: 0 255
+       @User: Standard */
+    AP_GROUPINFO("SPDREG", 5, AP_TTLServo, servo_des_run_speed_reg, RUNNING_SPEED_REG),
+
+    /* @Param: ID_BM
+       @DisplayName: Servo IDs bitmask
+       @Description: Bitmask of the servo IDs connected. Enable the servo in the
+                     corresponding servo_channel slot. Servo ID 0 corresponds to
+                     servo1_channel
+       @Bitmask: 0:ID 0, 1:ID 1, 2:ID 2, 3:ID 3, 4:ID 4, 5:ID 5, 6:ID 6, 7:ID 7, 8:ID 8, 9:ID 9, 10:ID 10, 11:ID 11, 12:ID 12, 13:ID 13, 14:ID 14, 15:ID 15, 16:ID 16, 17:ID 17, 18:ID 18, 19:ID 19, 20:ID 20, 21:ID 21, 22:ID 22, 23:ID 23, 24:ID 24, 25:ID 25, 26:ID 26, 27:ID 27, 28:ID 28, 29:ID 29, 30:ID 30, 31:ID 31
+       @User: Advanced */
+    AP_GROUPINFO("ID_BM", 6, AP_TTLServo, servo_id_mask, 0),
+
+    /* @Param: POSREG
+       @DisplayName: Servo goal position register address
+       @Description: Address of the servo's goal position register, to which the
+                     desired position will be written to
+       @Range: 0 255
+       @User: Standard */
+    AP_GROUPINFO("POSREG", 7, AP_TTLServo, servo_goal_pos_reg, GOAL_POSITION_REG),
+
+    AP_GROUPEND
+};
+
+// Constructor
+AP_TTLServo::AP_TTLServo(void){
+
+    // Set defaults from the parameter table
+    AP_Param::setup_object_defaults(this, var_info);
+}
+
+// Calculate communication protocol CRC (same as Robotis Dynamixel 1.0 protocol CRC)
+uint8_t AP_TTLServo::calculate_crc(uint8_t *txpacket, uint8_t len){
+    uint8_t checkSum, i;
+
+    for(i = PKT_ID, checkSum = 0; i < len; i++){
+        checkSum += txpacket[i];
+    }
+
+    return(~checkSum);
+
+}
+
+// Use a broadcast to set the speed of all servos.
+// Without speed configuration, servos will not run!
+void AP_TTLServo::configure_servos(void){
+
+    send_command(BROADCAST_ID, servo_des_run_speed_reg, servo_des_run_speed, 2);
+}
+
+// Use a broadcast ping to find attached servos
+void AP_TTLServo::detect_servos(void){
+    uint8_t txpacket[6];
+
+    txpacket[PKT_ID] = BROADCAST_ID;
+    txpacket[PKT_LENGTH] = 2;
+    txpacket[PKT_INSTRUCTION] = INST_PING;
```

Comment:
Use a structure.

Reply:
Using a structure here would mean using a structure in the send_packet function. Should I do that?

Reply:
Have a look in `AP_Hott_Telem.cpp` for an example.

Reply:
Done